src/classes/UTIL_OrgTelemetry.cls

```diff
@@ -0,0 +1,255 @@
+            edaSettings.Account_Processor__c != null && edaSettings.Account_Processor__c == UTIL_Describe.getAdminAccRecTypeID());
+        
+        featureManager.setPackageBooleanValue(TelemetryParameterName.HasDefaultHouseholdAccountModel.name(), 
+            edaSettings.Account_Processor__c != null && edaSettings.Account_Processor__c == UTIL_Describe.getHhAccRecTypeID());
+
+        featureManager.setPackageBooleanValue(TelemetryParameterName.IsEnabled_ErrorHandling.name(), edaSettings.Disable_Error_Handling__c == false);
+
+        featureManager.setPackageBooleanValue(TelemetryParameterName.HasCustomAdministrativeAccountNaming.name(), edaSettings.Admin_Account_Naming_Format__c == Label.acctNamingOther);
+
+        featureManager.setPackageBooleanValue(TelemetryParameterName.HasCustomHouseholdAccountNaming.name(), edaSettings.Household_Account_Naming_Format__c == Label.acctNamingOther);
+
+        featureManager.setPackageBooleanValue(TelemetryParameterName.HasValueInversionReciprocalMethod.name(), edaSettings.Reciprocal_Method__c == 'Value Inversion');
+
+        featureManager.setPackageBooleanValue(TelemetryParameterName.IsEnabled_StoreErrors.name(), edaSettings.Store_Errors_On__c);
+    }
+
+    /**
+     * @description Handles the calculation of all EDA feature telemetry that involves org-related
+     configuration.
+     */
+    private void handleOrgEnvironmentTelemetry() {
+        Integer activeCourseConnectionRecordTypes = 0;
+        Integer activeEthnicityPicklistValues = 0;
+        Boolean usingOldCourseDescription = false;
+
+        // Determine the number of active Course Connection record types.
+        Map<String, Schema.RecordTypeInfo> courseConnectionRecordTypesByDeveloperName = 
+                Schema.SObjectType.Course_Enrollment__c.getRecordTypeInfosByDeveloperName();
+        for (String developerName : courseConnectionRecordTypesByDeveloperName.keySet()) {
+            Schema.RecordTypeInfo rti = courseConnectionRecordTypesByDeveloperName.get(developerName);
+            if (rti.getDeveloperName() != 'Master' && rti.isActive()) {
+                activeCourseConnectionRecordTypes++;
+            }
+        }
+
+        // Determine the number of active Ethnicity picklist values.
+        Schema.DescribeFieldResult fieldResult = Contact.Ethnicity__c.getDescribe();
+        List<Schema.PicklistEntry> picklistEntries = fieldResult.getPicklistValues();
+        for (Schema.PicklistEntry pe : picklistEntries) {
+            if (pe.isActive()) {
+                activeEthnicityPicklistValues++;
+            }
+        }
+
+        // Determine if the customer is using the new Extended Description field on the Course object.
+        // For this, we look at the 500 most recently modified Course records.
+        List<Course__c> courses = [SELECT Description__c, Extended_Description__c FROM Course__c ORDER BY LastModifiedDate DESC LIMIT 500];
+        for (Course__c course : courses) {
```

Comment:
@lifewithryan Thanks for calling out this -- from the link you sent me, "Developers should always use a SOQL for loop to process query results that return many records, to avoid the limit on heap size." -- I hadn't really considered heap size, and since Extended Description is a long text area, maybe we would be better off doing the SOQL in the for loop.

Honestly I almost /never/ do SOQL in a for loop -- I don't really think I ever understood the reasoning behind it before. So thanks!

Reply:
Updated.